modules/clients/src/test/java/org/apache/ignite/internal/processors/rest/JettyRestProcessorSecurityCreateDestroyPermissionTest.java

```diff
@@ -0,0 +1,306 @@
+        assertNotNull(grid(0).cache(UNDECLARED_CACHE));
+    }
+
+    /** */
+    @Test
+    public void testDestroyCacheWithAdminPerms() throws Exception {
+        assertNotNull(grid(0).getOrCreateCache(CACHE_NAME));
+        assertNotNull(grid(0).getOrCreateCache(UNDECLARED_CACHE));
+
+        assertEquals(SUCCESS_STATUS,
+            jsonField(execute(CLNT_ADMIN, CACHE_NAME, GridRestCommand.DESTROY_CACHE), STATUS_FLD));
+        assertEquals(SUCCESS_STATUS,
+            jsonField(execute(CLNT_ADMIN, UNDECLARED_CACHE, GridRestCommand.DESTROY_CACHE), STATUS_FLD));
+
+        assertNull(grid(0).cache(CACHE_NAME));
+        assertNull(grid(0).cache(UNDECLARED_CACHE));
+    }
+
+    /** */
+    @Test
+    public void testDestroyCacheWithCachePermissions() throws Exception {
+        assertNotNull(grid(0).getOrCreateCache(CACHE_NAME));
+        assertNotNull(grid(0).getOrCreateCache(UNDECLARED_CACHE));
+
+        checkFailWithError(execute(UNALLOWED_CLNT_CACHE, CACHE_NAME, GridRestCommand.DESTROY_CACHE),
+            CACHE_DESTROY);
+        checkFailWithError(execute(UNALLOWED_CLNT_CACHE, UNDECLARED_CACHE, GridRestCommand.DESTROY_CACHE),
+            CACHE_DESTROY);
+
+        assertNotNull(grid(0).cache(CACHE_NAME));
+        assertNotNull(grid(0).cache(UNDECLARED_CACHE));
+
+        assertEquals(SUCCESS_STATUS,
+            jsonField(execute(ALLOWED_CLNT_CACHE, CACHE_NAME, GridRestCommand.DESTROY_CACHE), STATUS_FLD));
+
+        checkFailWithError(execute(ALLOWED_CLNT_CACHE, UNDECLARED_CACHE, GridRestCommand.DESTROY_CACHE),
+            CACHE_DESTROY);
+
+        assertNull(grid(0).cache(CACHE_NAME));
+
+        assertNotNull(grid(0).cache(UNDECLARED_CACHE));
+    }
+
+    /** */
+    @Test
+    public void testDestroyCacheWithSystemPermissions() throws Exception {
+        assertNotNull(grid(0).getOrCreateCache(CACHE_NAME));
+        assertNotNull(grid(0).getOrCreateCache(UNDECLARED_CACHE));
+
+        checkFailWithError(execute(UNALLOWED_CLNT_SYSTEM, CACHE_NAME, GridRestCommand.DESTROY_CACHE),
+            CACHE_DESTROY);
+        checkFailWithError(execute(UNALLOWED_CLNT_SYSTEM, UNDECLARED_CACHE, GridRestCommand.DESTROY_CACHE),
+            CACHE_DESTROY);
+
+        assertNotNull(grid(0).cache(CACHE_NAME));
+        assertNotNull(grid(0).cache(UNDECLARED_CACHE));
+
+        assertEquals(SUCCESS_STATUS,
+            jsonField(execute(ALLOWED_CLNT_SYSTEM, CACHE_NAME, GridRestCommand.DESTROY_CACHE), STATUS_FLD));
+        assertEquals(SUCCESS_STATUS,
+            jsonField(execute(ALLOWED_CLNT_SYSTEM, UNDECLARED_CACHE, GridRestCommand.DESTROY_CACHE), STATUS_FLD));
+
+        assertNull(grid(0).cache(CACHE_NAME));
+        assertNull(grid(0).cache(UNDECLARED_CACHE));
+    }
+
+    /**
+     * Execute REST command and return result.
+     *
+     * @param login Login.
+     * @param cacheName Cache name.
+     * @param cmd Command.
+     */
+    protected String execute(String login, String cacheName, GridRestCommand cmd) throws Exception {
```

Comment:
Could you extract this method to some **test utils** method? I suppose it would useful.